'i18n.merge_file' has been ignoring positional arguments for a time. Since 0.60.1 they are deprecated and an error since 0.61.0.

--- meld-3.21.0/data/meson.build.orig	2020-04-19 03:11:05.000000000 +0000
+++ meld-3.21.0/data/meson.build	2022-01-31 22:10:41.428528942 +0000
@@ -42,7 +42,6 @@
 appdata_conf.set('appid', application_id)
 
 appdata_file = i18n.merge_file(
-  'appdata',
   input: configure_file(
     input: files('org.gnome.meld.appdata.xml.in.in'),
     output: 'org.gnome.meld.appdata.xml.in',
